src/Orleans/Threading/CancellationTokenManager.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Orleans.Runtime;
+
+namespace Orleans.Threading
+{
+    internal class CancellationTokenManager
+    {
+        /// <summary>
+        /// Sets target activation to the found instances of type GrainCancellationToken
+        /// </summary>
+        /// <param name="arguments"> Grain method arguments list</param>
+        /// <param name="target"> Target grain reference</param>
+        public void SetGrainCancellationTokensTarget(object[] arguments, GrainReference target)
+        {
+            if (arguments == null) return;
+            for (var i = 0; i < arguments.Length; i++)
+            {
+                var argument = arguments[i];
+                if (argument is GrainCancellationToken)
+                {
+                    ((GrainCancellationToken) argument).TargetGrainReference = target;
+                }
+            }
```

Comment:
I haven't seen any difference on benchmarks, but since it is a hot path and 99% this check will return false - it's better to have 1 check than two for those who doesn't using `GrainCancellationToken`s.